src/adapter/src/coord/introspection.rs

```diff
@@ -86,24 +87,34 @@ pub fn auto_run_on_introspection<'a, 's>(
 pub fn check_cluster_restrictions(
     catalog: &impl SessionCatalog,
     plan: &Plan,
-    depends_on: &Vec<GlobalId>,
 ) -> Result<(), AdapterError> {
     // We only impose restrictions if the current cluster is the introspection cluster.
     let cluster = catalog.active_cluster();
     if cluster != MZ_INTROSPECTION_CLUSTER.name {
         return Ok(());
     }
 
-    // Allows explain queries.
-    if let Plan::Explain(_) = plan {
-        return Ok(());
-    }
+    // Only continue, and check restrictions, if a Plan would run some computation on the cluster.
+    //
+    // Note: We get the dependencies from the Plans themselves, because it's only after planning
+    // that we actually know what objects we'll need to reference.
```

Comment:
I think that we should open a GH issue and reference the issue in this comment.

Reply:
Will do!